tls/s2n_cipher_suites.c

```diff
@@ -1271,11 +1324,30 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
                 continue;
             }
 
+            /* Don't immediately choose the first negotiable cipher we find. ChaCha20 boosting may be enabled so we will need to continue iterating */
+            if (!server_preference_match) {
+                server_preference_match = match;
+            }
```

Comment:
Include this in the mutual_chacha20_boosting_support block. It should be clear we're doing this for the chacha20 use case. In most cases, we WILL still immediately choose the first negotiable cipher suite we find :)